components/Layout.tsx

```diff
@@ -1,15 +1,33 @@
-import React from 'react'
+import React, {useEffect} from 'react'
 import Header from './Header/Header'
+import {useUser} from "@auth0/nextjs-auth0";
+import {userProfileState} from "../atoms/userProfile";
+import {useSetRecoilState} from "recoil";
+import {getUserFromFirebase} from "../utils/helpers/userHelper";
 
 interface LayoutProps {
     children: React.ReactNode
 }
 
-const Layout = ({ children }: LayoutProps) => {
+
+const Layout = ({children}: LayoutProps) => {
+    const {user, isLoading} = useUser();
+    const setUser = useSetRecoilState(userProfileState);
```

Comment:
Actually, we need functions for both. But the above function call `getUserFromFirebase` returns the data from the user collection, not profile. So it is correct to store it in a state variable entitled user.

However, we should not be storing it in the `userProfileState` atom, we should store it in the new atom mentioned above.

Reply:
Do we need both? I believe all around the app we only use info from the profile. We used the profile to decouple basic user info from profile data. Part of this user info is now stored in Auth0. We use user collection to store, e.g., posts ids from the user. If we need this we can query it directly from Firebase.
Profile is less prone to changes so it can go in an atom. Other stuff it is better to fetch it in real time (cc @EandrewJones).

Reply:
No. You are right. I was thinking we'd need the user so we could easily access their id, but the simple solution is to store the hash for the user and profile as a field within the profile data. Then it's  always available from recoil.